src/webui/webapplication.cpp

```diff
@@ -299,13 +299,25 @@ const Http::Environment &WebApplication::env() const
     return m_env;
 }
 
+void WebApplication::logoutAllSessions()
+{
+	qDeleteAll(m_sessions);
+	m_sessions.clear();
+}
+
 void WebApplication::setUsername(const QString &username)
 {
+    if (!m_username.isEmpty() && (m_username != username))
+        logoutAllSessions();
+    m_username = username;
     m_authController->setUsername(username);
 }
 
 void WebApplication::setPasswordHash(const QByteArray &passwordHash)
 {
+    if (!m_passwordHash.isEmpty() && (m_passwordHash != passwordHash))
+        logoutAllSessions();
+    m_passwordHash = passwordHash;
     m_authController->setPasswordHash(passwordHash);
```

Comment:
Why not just:
```suggestion
    m_authController->setPasswordHash(passwordHash);
    logoutAllSessions();
```
and the same to `WebApplication::setUsername()`.

I don't really like it storing `m_username` & `m_passwordHash` unless truly necessary due to their sensitive context.

Reply:
> I don't really like it storing `m_username` & `m_passwordHash` due to their sensitive context.

They are stored in `AuthController` anyway so they actually share the same data due to Qt implicitly sharing.

Apparently, the author does not want the sessions to be terminated if the username/password is set to the same value. On the other hand, I think this case could be handled from the caller side.

Reply:
>Apparently, the author does not want the sessions to be terminated if the username/password is set to the same value.

I fail to understand it. Why submit username or password if it is not intended to be changed? If someone intend username (or password) change then it should be ok to just remove all sessions.
Furthermore, the password hash is not going to be the same even for the same password. So the equal comparison seems useless anyway.

>They are stored in AuthController anyway so they actually share the same data due to Qt implicitly sharing.

In any case, we should strive to minimize them.

>On the other hand, I think this case could be handled from the caller side.

Yes. IIRC, webui won't submit username/password updates unless user provided them. The webapi users should either not blindly submit them too.

Reply:
> IIRC, webui won't submit username/password updates unless user provided them. The webapi users should not blindly submit them too.

It's not about webui/webapi submitting an unchanged username/password, but about `WebUI::configure` being triggered every time any prefs changed.

https://github.com/qbittorrent/qBittorrent/blob/1c43286616e5c1a6420f9d1049ae860dbe0c2d6f/src/webui/webui.cpp#L100-L101

Reply:
>but about WebUI::configure being triggered every time any prefs changed.

It should be done there instead.

Reply:
>Furthermore, the password hash is not going to be the same even for the same password.

Apparently, you wanted to say that the same hash can correspond to different passwords.

Reply:
> > Apparently, you wanted to say that the same hash can correspond to different passwords.
> 
> No. I meant the hashes are different even if the same password is passed to the hash generator.

Then how do you verify the password?

Reply:
>Then how do you verify the password?

User provide the raw password and hash it again with the same hash parameters as the previous one.

Reply:
> User provide the raw password and hash it again with the same hash parameters as the previous one. And see whether they are the same.

Yes. So we always have the same hash for the same password in discussed code, no?